dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -56,22 +63,56 @@ class CompanyDataControllerTest {
 
     @Test
     fun `post a dummy company and check if that specific company can be queried by its name`() {
-        val testCompanyInformation = testDataProviderForEuTaxonomyDataForNonFinancials
-            .getCompanyInformationWithoutIdentifiers(1).first()
-        tokenHandler.obtainTokenForUserType(TokenHandler.UserType.Admin)
-        val postCompanyResponse = companyDataControllerApi.postCompany(testCompanyInformation)
+        val storedCompany = postFirstCompanyWithoutIdentifiers()
         tokenHandler.obtainTokenForUserType(TokenHandler.UserType.SomeUser)
         val getCompaniesByNameResponse = companyDataControllerApi.getCompanies(
-            searchString = testCompanyInformation.companyName,
+            searchString = storedCompany.companyInformation.companyName,
             onlyCompanyNames = true,
         )
-        val expectedCompany = StoredCompany(postCompanyResponse.companyId, testCompanyInformation, emptyList())
+        val expectedCompany = StoredCompany(storedCompany.companyId, storedCompany.companyInformation, emptyList())
         assertTrue(
             getCompaniesByNameResponse.contains(expectedCompany),
             "Dataland does not contain the posted company."
         )
     }
 
+    @Test
+    fun `post two dummy companies and check if the distinct endpoint returns all values`() {
+        tokenHandler.obtainTokenForUserType(TokenHandler.UserType.Admin)
+        val testCompanyInformation = testDataProviderForEuTaxonomyDataForNonFinancials
+            .getCompanyInformationWithoutIdentifiers(2)
+        testCompanyInformation.forEach {
+            companyDataControllerApi.postCompany(it)
+        }
+        val distinctValues = companyDataControllerApi.getAvailableCompanySearchFilters()
+        assertTrue(distinctValues.sectors?.containsAll(testCompanyInformation.map { it.sector }) ?: false)
+        assertTrue(distinctValues.countryCodes?.containsAll(testCompanyInformation.map { it.countryCode }) ?: false)
```

Comment:
Ich würde vorschlagen, sectors und countryCodes hier mit !! eher non-nullable zu machen.
Grund: Wenn diese beiden Listen null sind, dann stimmt etwas Grundsätzliches nicht, und dann würde ich gerne eine Nullpointer-Exception sehen.
Wenn ich sie hingegen nullable mache, und in diesem Fall hier einfach "false" ausgeben lasse, werde ich in dem failenden Test ja nur sehen, dass der AssertTrue einfach nicht erfolgreich war.
Ich würden dann nicht quasi direkt sehen, dass dieses AsserTrue von Grund auf gar keine Chance hat, erfüllt zu werden, weil die Listen fehlen.

Reply:
Mit Stephan besprochen und auf non-nullable gesetzt